### PR TITLE
add vertexAttribIPointer support Int Data Type

### PR DESCRIPTION
**Pull request by GoodLuckIce**
add createVertexBuffer support  Int  Data Type

### CODE REVIEW

**Comment by sebavan** [comment]
This does not compile as you need to put vertexAttribIPointer definition in this file https://github.com/sebavan/Babylon.js/blob/master/packages/dev/core/src/LibDeclarations/webgl2.d.ts#L1

**Review by deltakosh — APPROVED** [review]

**Review by RaananW — CHANGES_REQUESTED** [review]
Please run the formatter after you are done with the changes (or follow the same formatting as the rest of the code)

File: packages/dev/core/src/Engines/thinEngine.ts [hunk 1]
```diff
@@ -2240,6 +2240,11 @@ export class ThinEngine {
         if (changed || this._vaoRecordInProgress) {
             this.bindArrayBuffer(buffer);
             this._gl.vertexAttribPointer(indx, size, type, normalized, stride, offset);
+            if (type === this._gl.UNSIGNED_INT || type === this._gl.INT) {
+                this._gl.vertexAttribIPointer(indx, size, type, stride, offset);
```
[the remark this answers is not in the document]
Comment (sebavan) [reply packages/dev/core/src/Engines/thinEngine.ts:2244]: The definition should be added intohttps://github.com/sebavan/Babylon.js/blob/master/packages/dev/core/src/LibDeclarations/webgl2.d.ts#L1 as we do for the rest of webgl2 specific functions instead.

**Comment by sebavan** [comment]
closed for https://github.com/BabylonJS/Babylon.js/pull/12830 so I can merge for the next release

**Pull request closed without merging**